core/crypto/aes-gcm-sw.c

```diff
@@ -38,14 +38,16 @@ void internal_aes_gcm_ghash_update(struct internal_aes_gcm_state *state,
 				   const void *head, const void *data,
 				   size_t num_blocks)
 {
-	size_t n = 0;
-
 	if (head)
 		ghash_update_block(state, head);
 
-	for (n = 0; n < num_blocks; n++)
-		ghash_update_block(state,
-				   (uint8_t *)data + n * TEE_AES_BLOCK_SIZE);
+	if (data) {
+		size_t n = 0;
+
+		for (; n < num_blocks; n++)
+			ghash_update_block(state,
+					   (const uint8_t *)data + n * TEE_AES_BLOCK_SIZE);
```

Comment:
Must split this line that exceeds 80 charaters.

minor: I thinnk `const` is not mandated here.

You could simplify the patch using:
```patch
 	if (head)
 		ghash_update_block(state, head);
 
+	if (!data)
+		return;
 
 	for (n = 0; n < num_blocks; n++)
 		ghash_update_block(state,
 				   (uint8_t *)data + n * TEE_AES_BLOCK_SIZE);
```

Reply:
I kept the const since there is no need to cast constness away.

I think it is good to avoid returns in the middle fo a function.

Id be willing to extract the data part to a static internal_aes_gcm_ghash_update_data_blocks() function in order to keep linelength down if it is not ok as is...

Reply:
> I think it is good to avoid returns in the middle of a function.

I think it's good to avoid complicated code in order to avoid returning in the middle of a function.
In this case it's not a big deal, but what @etienne-lms is suggesting is preferable and consistent with the style used in this project.

Reply:
ok fine.
For the 80 char per line convention, prefer:
```c
	if (data)
		for (n = 0; n < num_blocks; n++)
			ghash_update_block(state,
					   (const uint8_t *)data +
					   n * TEE_AES_BLOCK_SIZE);
```
Note: braces not required for the `if ()` block.